File: app/src/main/java/ir/apptune/antispam/MainActivity.java
package ir.apptune.antispam;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;

import ir.apptune.antispam.fragments.MyFragment;

public class MainActivity extends AppCompatActivity {
    private Toolbar toolbar;
    private ViewPager mPager;
    private TabLayout mTabs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        toolbar = (Toolbar) findViewById(R.id.app_bar);
        toolbar.setTitle("");
        setSupportActionBar(toolbar);

        mPager = findViewById(R.id.pager);
        mTabs = findViewById(R.id.tabs);
        mPager.setAdapter(new MyPagerAdapter(getSupportFragmentManager()));
        mTabs.setupWithViewPager(mPager);
        startService(new Intent(this, ShakerService.class));

    }

    class MyPagerAdapter extends FragmentPagerAdapter {

        String[] tabs;

        public MyPagerAdapter(FragmentManager fm) {
            super(fm);
            tabs = getResources().getStringArray(R.array.tabs);
        }

        @Override
        public Fragment getItem(int position) {
            MyFragment myFragment = MyFragment.getInstance(position);
            return myFragment;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return tabs[position];
        }

        @Override
        public int getCount() {
            return 2;
        }
    }


}

File: app/src/main/java/ir/apptune/antispam/MainActivity.kt
package ir.apptune.antispam

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.core.content.ContextCompat
import androidx.navigation.NavController
import androidx.navigation.NavDestination
import androidx.navigation.findNavController
import androidx.navigation.ui.setupWithNavController
import ir.apptune.antispam.databinding.ActivityMainBinding
import ir.apptune.antispam.features.service.ForegroundService
import ir.apptune.antispam.utils.getStatusBarHeight

/**
 * We use SingleActivity approach
 *
 */
class MainActivity : AppCompatActivity() {

    private val binding by lazy { ActivityMainBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        ContextCompat.startForegroundService(this, Intent(this, ForegroundService::class.java))
        setUpBottomNav()
        setSupportActionBar(binding.toolbar)
        setStatusBar()
    }

    private fun setUpBottomNav() {
        binding.bottomNav.setupWithNavController(findNavController(R.id.navFragment))
        findNavController(R.id.navFragment).addOnDestinationChangedListener { _: NavController, navDestination: NavDestination, _: Bundle? ->
            when (navDestination.id) {
                R.id.splashFragment -> hideToolbarBottomNav()
                R.id.introFragment -> hideBottomNav()
                else -> showToolbarBottomNav()
            }
        }
    }

    private fun hideBottomNav() = with(binding) {
        bottomNav.visibility = View.GONE
        toolbar.visibility = View.VISIBLE
    }

    private fun showToolbarBottomNav() = with(binding) {
        bottomNav.visibility = View.VISIBLE
        toolbar.visibility = View.VISIBLE
    }

    private fun hideToolbarBottomNav() = with(binding) {
        bottomNav.visibility = View.GONE
        toolbar.visibility = View.GONE
    }

    private fun setStatusBar() = with(binding) {
        val statusBarHeight = getStatusBarHeight(this@MainActivity)
        viewStatusBar.layoutParams =
            ConstraintLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, statusBarHeight)
        viewStatusBar.background =
            ContextCompat.getDrawable(this@MainActivity, R.drawable.toolbar_shape)
    }
}
